adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceRtbInterstitialAd.m

```diff
@@ -0,0 +1,193 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterIronSourceRtbInterstitialAd.h"
+#import "GADMAdapterIronSourceConstants.h"
+#import "GADMAdapterIronSourceInterstitialAdDelegate.h"
+#import "GADMAdapterIronSourceUtils.h"
+#import "GADMediationAdapterIronSource.h"
+
+
+
+@interface IronSourceRtbInterstitialAdDelegate : NSObject
+@property (nonatomic,   weak) GADMAdapterIronSourceRtbInterstitialAd *parentAdapter;
+@property (nonatomic, strong) GADMediationInterstitialLoadCompletionHandler delegate;
+@property (nonatomic, strong) id<GADMediationInterstitialAdEventDelegate>  gadIsDelegate;
+
+
+
+- (instancetype)initWithParentAdapter:(GADMAdapterIronSourceRtbInterstitialAd *)parentAdapter andNotify:(GADMediationInterstitialLoadCompletionHandler)delegate;
+@end
+
+@interface GADMAdapterIronSourceRtbInterstitialAd ()
+
+// An ad event delegate to invoke when ad rendering events occur.
+@property(weak, nonatomic) id<GADMediationInterstitialAdEventDelegate> interstitialAdEventDelegate;
+
+/// Holds the ID of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceID;
+
+/// Holds the state of the ad instance to be presented.
+@property (nonatomic, strong) IronSourceRtbInterstitialAdDelegate *biddingInterstitialAdDelegate;
+
+@end
+
+
+@implementation GADMAdapterIronSourceRtbInterstitialAd
+
+#pragma mark InstanceMap and Delegate initialization
+// The class-level delegate handling callbacks for all instances
+
+
+#pragma mark - Load functionality
+
+- (void)loadInterstitialForAdConfiguration:
+(GADMediationInterstitialAdConfiguration *)adConfiguration
+                         completionHandler:
+(GADMediationInterstitialLoadCompletionHandler)completionHandler {
+    _interstitalAdLoadCompletionHandler = completionHandler;
+    
+    NSDictionary *credentials = [adConfiguration.credentials settings];
+    NSString *applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    
+    if (applicationKey != nil && ![GADMAdapterIronSourceUtils isEmpty:applicationKey]) {
+        applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    } else {
+        NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                          GADMAdapterIronSourceErrorInvalidServerParameters,
+                                                                          @"Missing or invalid IronSource application key.");
+        
+        _interstitalAdLoadCompletionHandler(nil, error);
+        return;
+    }
```

Comment:
I see that applicationKey is NOT used for loading the ad. So, I think checking for applicationKey can be removed.

Reply:
Thanks! updated